dev/test/test_dataloader.c

```diff
@@ -0,0 +1,299 @@
+/*
+Tests our DataLoader
+
+compile and run as (from dev/test directory)
+gcc -O3 -I../../llmc -o test_dataloader test_dataloader.c -lm && ./test_dataloader
+
+*/
+
+#include "../../llmc/dataloader.h"
+
+char shard_name[64];
+int num_tokens = 140;
```

Comment:
nit: maybe upper case these global vars in order to have an easier time distinguishing them from locally defined vars